frontend/src/metabase/common/components/DataPicker/components/DataPickerModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { useCallback } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+
+import type { CollectionId, TableId } from "metabase-types/api";
+
+import type { EntityPickerModalOptions, EntityTab } from "../../EntityPicker";
+import { EntityPickerModal, defaultOptions } from "../../EntityPicker";
+import type { NotebookDataPickerValueItem, TablePickerValue } from "../types";
+
+import { TablePicker } from "./TablePicker";
+
+interface Props {
+  /**
+   * TODO: use this prop in https://github.com/metabase/metabase/issues/40719
+   */
+  collectionId: CollectionId | null | undefined;
+  value: TablePickerValue | null;
+  onChange: (value: TableId) => void;
+  onClose: () => void;
+}
+
+const options: EntityPickerModalOptions = {
+  ...defaultOptions,
+  hasConfirmButtons: false,
+};
+
+export const DataPickerModal = ({ value, onChange, onClose }: Props) => {
+  const handleItemChange = useCallback(
+    (item: NotebookDataPickerValueItem) => {
+      onChange(item.id);
+      onClose();
+    },
+    [onChange, onClose],
+  );
+
+  const tabs: [
+    EntityTab<NotebookDataPickerValueItem["model"]>,
```

Comment:
can you please clarify this type? It's not clear why do we expect a tuple with a type and destructured type array tbh

Reply:
it's because we always expect at least one tab

Reply:
I'm not a big fan of using types to enforce this.
Actually, I'm not a big fan of enforcing this at all :upside_down_face: 

:heavy_check_mark:  Simplified: 470c461b7959bf3b50276f615725b346efff17b1